lisa/util/__init__.py

```diff
@@ -352,6 +356,32 @@ def enable(self) -> None:
         self._switch(True)
 
 
+class MutexObject(Generic[T]):
+    """
+    This is a wrapper class which can be used to perform actions of any objects in
+    atomic manner. This is helpful in synchronization when a single object is shared
+    with multiple threads.
+    For example,
+        obj = MutexObject(someObject)
+        with obj.get_object() as mutexobj:
+            mutexobj.method()
+
+    This is same as doing someObject.method() atomically.
+    """
+
+    __object: T
+    __lock: Lock
+
+    def __init__(self, obj: T) -> None:
+        self.__object = obj
+        self.__lock = Lock()
+
+    @contextmanager
+    def get_object(self) -> Generator[T, None, None]:
```

Comment:
IMHO `lock` would be a better name.

Reply:
How about `get_locked_object`? Since it returns the actual object and not lock.

Reply:
`lock` sounds better. change it.